subt_ign/launch/cave_circuit.ign

```diff
@@ -1191,27 +1386,95 @@
+            robotName = "#{robotType} #{mUAVStr}Config #{robotConfigN}"
           else
             robotName = config
           end
           modelURI = "#{fuelPrefix}/#{robotName}"
         end
         robotSpawned += 1
-%>
-<%=     spawner(name, modelURI, $worldName, posX, posY, spawnWorldZPos, 0, 0, spawnWorldYaw) %>
-<%=     rosExecutables(name, $worldName) %>
-<%
+
+        spawnStringTmp = spawner(name, modelURI, $worldName, robot.pos.x, robot.pos.y, robot.pos.z, 0, 0, robot.rot.yaw)
+        executables = rosExecutables(name, $worldName)
       end
     end
+
+    allExecutables += executables
+
+    if marsupialParents.has_key?(name)
+      childModelName = marsupialParents[name]
+       detachableJointPlugin = REXML::Document.new <<-HEREDOC
+         <plugin filename="libignition-gazebo-detachable-joint-system.so" name="ignition::gazebo::systems::DetachableJoint">
+         <parent_link>base_link</parent_link>
+         <child_model>#{childModelName}</child_model>
+         <child_link>base_link</child_link>
+         <topic>/model/#{childModelName}/detach</topic>
+         <suppress_child_warning>true</suppress_child_warning>
+         </plugin>
+       HEREDOC
+
+       # This will be use to attach the drone platform to the base.
+       platformJointPlugin = REXML::Document.new <<-HEREDOC
+         <plugin filename="libignition-gazebo-detachable-joint-system.so" name="ignition::gazebo::systems::DetachableJoint">
+         <parent_link>base_link</parent_link>
+         <child_model>#{name}_platform</child_model>
+         <child_link>base_link</child_link>
+         <suppress_child_warning>true</suppress_child_warning>
+         </plugin>
+       HEREDOC
+
+       rosExecutable = <<-HEREDOC
+         <executable name="#{childModelName}_marsupial">
+           <command>roslaunch --wait subt_ros marsupial_topics.launch name:=#{childModelName}</command>
+         </executable>
+       HEREDOC
+       allExecutables += rosExecutable
+
+       # This will be used to spawn the drone platform.
+       platform = <<-HEREDOC
+         <spawn name="#{name}_platform">
+           <name>#{name}_platform</name>
+           <allow_renaming>false</allow_renaming>
+           <pose>#{robot.pos.x + 0.078} #{robot.pos.y} #{robot.pos.z-0.15} 0 0 0</pose>
+           <world>#{$worldName}</world>
+           <is_performer>false</is_performer>
+           <sdf version='1.6'>
+             <include>
+               <uri>https://fuel.ignitionrobotics.org/1.0/openrobotics/models/DronePlatformX1</uri> 
+             </include>
+           </sdf>
+         </spawn>
+       HEREDOC
+
+       # Wrap the temporary spawn string with a root element. This is needed
+       # because the xml parser needs a root element. When we're adding the
+       # plugin, we print only the contents of the root element.
+       spawnXml = REXML::Document.new("<ignition>#{spawnStringTmp}</ignition>")
+       search = "//spawn[@name='#{name}']"
```

Comment:
Submitted models don't use the new `<spawn>` tag. Can we also check for the `ignition::launch::GazeboFactory` here?

Reply:
Actually, we probably need to convert them to `<spawn>` tags either in the `spawner.rb` scripts or by parsing the result of the `spawner` function on line 1402.

Reply:
I think I converted the submitted models in: f13f1a12c620541d5c8c0acb1bde18fe1755bd7d